File: 2058-ConcatenationOfArray/2058-ConcatenationOfArray.java
// Last updated: 4/15/2026, 11:49:40 PM
class Solution {
    public int[] getConcatenation(int[] nums) {
        int array_len = nums.length;
        int[] newArr = new int[array_len * 2];
        for(int i = 0; i < nums.length; i++){
            newArr[i] = nums[i];
            newArr[i + nums.length] = nums[i];
        }
        return newArr;
    }
}

File: 2058-ConcatenationOfArray/2058-ConcatenationOfArray.cpp
// Last updated: 4/15/2026, 11:49:43 PM
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> myVector(nums.size() * 2, 0);
        for (int i = 0; i < nums.size(); ++i) {
            myVector[i] = nums[i];       // First copy
            myVector[i + nums.size()] = nums[i];   // Second copy
        }
        cout << endl;
        return myVector;  

    }
};